### ISSUE TITLE
Add sPDC method

### ISSUE THREAD

**Issue by dongqunxi** [scot-dev/scot#40]
Hi @kazemakase ,
I need to use the modified version of PDC, squared PDC, but I can not find it in the causal methods list. Can you give me some tips? Thanks!

**Comment by agramfort** [comment]
@kazemakase your tugraz email address does not work. How can I reach you? github was my only option...

**Comment by dongqunxi** [comment]
@kazemakase @agramfort Thanks for your helping on this issue. I edit the PDC function in https://gist.github.com/agramfort/9875439. And get the sPDC function, I am not sure if this modifying is opporiate, look forward to your suggestions, thanks!
```ruby
def sPDC(A, sigma=None, n_fft=None):
    """squared Partial directed coherence (sPDC)

    Parameters
    ----------
    A : ndarray, shape (p, N, N)
        The AR coefficients where N is the number of signals
        and p the order of the model.
    sigma : array, shape (N,)
        The noise for each time series.
    n_fft : int
        The length of the FFT.

    Returns
    -------
    P : ndarray, shape (n_fft, N, N)
        The estimated PDC.
    """
    p, N, N = A.shape

    if n_fft is None:
        n_fft = max(int(2 ** math.ceil(np.log2(p))), 512)

    H, freqs = spectral_density(A, n_fft)
    P = np.zeros((n_fft, N, N))

    if sigma is None:
        sigma = np.ones(N)

    for i in range(n_fft):
        B = H[i]
        B = np.abs(B) ** 2
        P[i] = B / np.sum(B, axis=0, keepdims=True)

    return P, freqs
```

**Comment by kazemakase** [comment]
I think you forgot to invert H (i.e.`B = linalg.inv(H[i])`).
Otherwise it looks good to me.

**Comment by dongqunxi** [comment]
@kazemakase Thanks a lot!

**Comment by cle1109** [comment]
@kazemakase should we add this measure?

**Comment by kazemakase** [comment]
Yes, I'd say so... that's why I left the issue open :)

**Comment by cle1109** [comment]
I'd like to implement sPDC so that we can finally close this issue. I was wondering if the following implementation would do the trick (instead of the one suggested above, which looks much more complicated):

    @memoize
    def sPDC(self):
        """Squared partial directed coherence

        .. math:: \mathrm{sPDC}_{ij}(f) = \\frac{\left| A_{ij}(f) \right|^2}{\sqrt{A_{:j}'(f) A_{:j}(f)}}  # TODO: fix equation

        References
        ----------
        L. Astolfi, F. Cincotti, D. Mattia, M. G. Marciani, L. Baccala,
        F. D. Fallani, S. Salinari, M. Ursino, M. Zavaglia, F. Babiloni. Partial
        directed coherence: a new concept in neural structure determination. IEEE
        Trans. Biomed. Eng. 53(9):1802-1812, 2006.
        """
        A = self.A()
        A = np.abs(A)**2
        return A / np.sum(A, axis=0, keepdims=True)

**Comment by kazemakase** [comment]
Looks good, but we should check the reference.

**Comment by cle1109** [comment]
I checked the [reference](http://www.sciencedirect.com/science/article/pii/S0010482511001272) mentioned by @dongqunxi because I don't have access to the IEEE one. The equation there is exactly what you found elsewhere.

Can the equation in the docstring be written without the sum?

**Comment by kazemakase** [comment]
What about writing it as a dot productt: `sum_k |A_kj(f)|² = 1'|A_:j(f)|²`

Sorry, can't express it betten in ASCII. 1' is supposed to be a row vector [1,1,...,1]

**Comment by cle1109** [comment]
.. math:: \mathrm{sPDC}_{ij}(f) = \\frac{\left| A_{ij}(f) \right|^2}{\mathbf{1} \left| A_{:j}(f) \right|^2}

?

**Comment by kazemakase** [comment]
Seems to look good, as far i can tell from plain Latex :)

Maybe transpose the 1-vector.

**Comment by cle1109** [comment]
But isn't the 1-vector supposed to be a row vector? So why transpose?

**Comment by kazemakase** [comment]
Because by default vectors are column vectors

**Comment by cle1109** [comment]
OK - just out of curiosity, do you have a reference for that? I always thought that there is no agreed upon definition.

**Comment by kazemakase** [comment]
No hard reference, but most texts assume row vectors.
For example you will usually find written that `Ax=b`, which only works if x and b are column vectors. Otherwise you'd have to write `xA=b`, which virtually nobody does...

**Comment by cle1109** [comment]
OK thanks. Let me know if you think we can merge #67.

**Comment by kazemakase** [comment]
Perhaps we should add a test.

**Comment by cle1109** [comment]
Resolved by #67.
